api_server/src/lib.rs

```diff
@@ -115,31 +111,35 @@ impl ApiServer {
                 .add(delta_us as usize);
         }
 
-        let http: Http<hyper::Chunk> = Http::new();
-
+        let http = Http::new();
         let f = listener
             .incoming()
-            .for_each(|(stream, _)| {
+            .for_each(move |stream| {
                 // For the sake of clarity: when we use self.efd.clone(), the intent is to
-                // clone the wrapping Rc, not the EventFd itself.
+                // clone the wrapping Arc, not the EventFd itself.
                 let service = ApiServerHttpService::new(
                     self.mmds_info.clone(),
                     self.vmm_shared_info.clone(),
                     self.api_request_sender.clone(),
                     self.efd.clone(),
                 );
-                let connection = http.serve_connection(stream, service);
-                // todo: is spawn() any better/worse than execute()?
-                // We have to adjust the future item and error, to fit spawn()'s definition.
-                handle.spawn(connection.map(|_| ()).map_err(|_| ()));
+                
+                let conn = http
+                    .serve_connection(stream, service)
+                    .map_err(|e| {
+                        eprintln!("server connection error: {}", e);
```

Comment:
We should use the Firecracker `logger` crate instead of something like `eprintln!` to log any kind of message.

Reply:
Can fix.

When https://github.com/hawkw/tokio-trace-prototype stabilizes/is released, we'd be grateful if you can test and provide feedback: https://github.com/hawkw/tokio-trace-prototype. It's a low-latency instrumentation/tracing system.